### PR TITLE
Switch to external `history-tree` library

### LINKED ISSUE #638: Move history tree to a separate repository
Or shall we move it to cl-containers?
If the latter, then we'd need to adapt the library to reuse the generic functions of cl-containers.
It's a bit more work, but this saves the community of an extra lib.

### PR DESCRIPTION
**Pull request by Ambrevar**
# Description

This moves the internal `history-tree` lib to https://github.com/atlas-engineer/history-tree/.

Fixes #638.

# To do:

- [x] Release 0.1.0
- [ ] Update submodule.
- [x] Package for Guix.
- [x] Document how to be use the history tree without globality.

# Checklist:
Everything in this checklist is required for each PR.  Please do not approve a PR that does not have all of these items.

- [x] I have pulled from master before submitting this PR
- [x] There are no merge conflicts.
- [x] I've added the new dependencies as:
  - [ ] ASDF dependencies,
  - [ ] Git submodules,
    ```sh
	cd /path/to/nyxt/checkout
    git submodule add https://gitlab.common-lisp.net/nyxt/py-configparser _build/py-configparser
    ```
  - [ ] and Guix dependencies.
- [x] My code follows the style guidelines for Common Lisp code. See:
  - [Norvig & Pitman's Tutorial on Good Lisp Programming Style (PDF)](https://www.cs.umd.edu/~nau/cmsc421/norvig-lisp-style.pdf)
  - [Google Common Lisp Style Guide](https://google.github.io/styleguide/lispguide.xml)
- [x] I have performed a self-review of my own code.
- [ ] My code has been reviewed by at least one peer.  (The peer review to approve a PR counts.  The reviewer must download and test the code.)
- [x] Documentation:
  - [ ] All my code has docstrings and `:documentation`s written in the aforementioned style.  (It's OK to skip the docstring for really trivial parts.)
  - [ ] I have updated the existing documentation to match my changes.
  - [ ] I have commented my code in hard-to-understand areas.
  - [ ] I have updated the `changelog.lisp` with my changes if it's anything user-facing (new features, important bug fix, compatibility breakage).
  - [ ] I have added a `migration.lisp` entry for all compatibility-breaking changes.
  - [ ] (If this changes something about the features showcased on Nyxt website) I have these changes described in the new/existing article at Nyxt website or will notify one of maintainters to do so.
- [x] Compilation and tests:
  - [ ] My changes generate no new warnings.
  - [ ] I have added tests that prove my fix is effective or that my feature works.  (If possible.)
  - [ ] New and existing unit tests pass locally with my changes.

### CODE REVIEW

**Comment by aadcg** [comment]
Nice! Has it landed on Guix already?

**Comment by aartaka** [comment]
Maybe name it to `ntree` then? Even though it's called `history-tree` in my BA thesis, no one read that and won't be offended :P

**Comment by Ambrevar** [comment]
I like `history-tree` because well, it describes exactly what it is and does.

`ntree` is less clear: it's a tree, but what is `n`?

**Comment by aadcg** [comment]
`tree` is indeed too generic. If you're bent on adding `n`, then I would suggest `nhistory-tree` or even `nhistory`.

**Comment by Ambrevar** [comment]
In my opinion we don't have to prefix everything with `n`.  It makes sense to prefix with `n` to disambiguate common words (like `nfiles`), but in other cases there is no need.  For instance we have cl-readability.

**Comment by aadcg** [comment]
All good, no strong opinion!

**Comment by Ambrevar** [comment]
@jmercouris I (at last!) added the option to disable history globality, thus reverting to the history structure we had before the GHT overhaul, thus answering issue #1677.
Does it fit the bill in your opinion?

Sorry for taking so long to do this considering it was a one-liner :p

**Comment by Ambrevar** [comment]
@aadcg I've released history-tree 0.1.0. You said you wanted to package it for Guix. Still want to do it? Let me know, I can also do it.
https://github.com/atlas-engineer/history-tree/

**Comment by aadcg** [comment]
Yes, I can do it.

File: source/buffer.lisp [hunk 1]
```diff
@@ -461,6 +461,17 @@ Suggestions are computed by the default search engine.")
     :type symbol
     :documentation "Select a download engine to use, such as `:lisp' or
 `:renderer'.")
+   (global-history-p
+    t
```
Comment (aadcg) [inline source/buffer.lisp:465]: My understanding is that @jmercouris asked to set this option to `nil` by default.

**Comment by aadcg** [comment]
I've sent the patch.

**Comment by Ambrevar** [comment]
And it's merged, thanks!

**Comment by aadcg** [comment]
Yes, I was going to tell you about it right now!

File: source/buffer.lisp (hunk 1 above)
Comment (Ambrevar) [reply source/buffer.lisp:465]: Well @jmercouris wanted at least the possibility to toggle globality.  Now which one should be the default?  Good question!  @jmercouris @aartaka ?

**Comment by Ambrevar** [comment]
Hehe!

Mostly ready to merge!

File: source/buffer.lisp (hunk 1 above)
Comment (aadcg) [reply source/buffer.lisp:465]: See #1677.

**Comment by aadcg** [comment]
We have issues tagged with the `history` label. Do we need to move them to the new repository?

**Comment by Ambrevar** [comment]
I just reviewed them and I believe they are all Nyxt-specific, they don't apply to the GHT per se.

**Comment by Ambrevar** [comment]
Let's merge then?

**Comment by aadcg** [comment]
I'm waiting on @jmercouris. On the call with me, he expressed that he wants `global-history-p` set to `nil` by default.

I'd prefer to merge it after he comments on this PR.

File: source/buffer.lisp (hunk 1 above)
Comment (aadcg) [reply source/buffer.lisp:465]: Friendly ping @jmercouris.

**Comment by aadcg** [comment]
I'll merge this when I talk to @jmercouris tomorrow.

**Comment by aadcg** [comment]
Thanks @ambrevar!

**Pull request merged**